src/Microsoft.OData.Core/RawValueWriter.cs

```diff
@@ -86,6 +90,21 @@ public void Dispose()
             this.textWriter = null;
         }
 
+#if NETSTANDARD2_0 || NETCOREAPP3_1_OR_GREATER
+        public ValueTask DisposeAsync()
+        {
+            return DisposeAsyncCore();
+        }
+
+        private async ValueTask DisposeAsyncCore()
+        {
+            Debug.Assert(this.textWriter != null, "The text writer has not been initialized yet.");
+
+            await this.textWriter.FlushAsync().ConfigureAwait(false);
+            this.textWriter = null;

```

Comment:
Why aren't we calling `this.textWriter.DisposeAsync()` here but we're disposing it in the sync version?

Reply:
+1

Reply:
@habbes @KenitoInc Fixed this by calling `DisposeAsync` method that internally calls `FlushAsync`. Nice catch